src/dsync/dsync.c

```diff
@@ -1173,9 +1188,14 @@ static void dsync_strmap_compare(mfu_flist src_list,
     mfu_flist dst_remove_list = MFU_FLIST_NULL; 
     mfu_flist src_cp_list     = MFU_FLIST_NULL; 
 
-    /* create dst remove list if sync option is on */
-    dst_remove_list = mfu_flist_subset(dst_list);
-    src_cp_list = mfu_flist_subset(src_list);
+    if (mfu_copy_opts->do_sync) {
+        /* create dst remove list if sync option is on */
+        if (options.delete) {
+            dst_remove_list = mfu_flist_subset(dst_list);
```

Comment:
@KnightKu It looks like the dst_remove_list is only being created in the delete option case now? If so, I would think the dst_remove_list also needs to be created when a regular sync is done.

Reply:
Hi @dsikich, I don't think we need dst_remove_list if delete option is disabled. You know we are using compare expression, so we can get the "extraneous files in dest" via EXIST=ONLY_DEST now, and dst_remove_list here is only used to remove the extraneous files if delete option enabled.